moz_ingest/io_modules/decoders/moz_ingest/common.lua

```diff
@@ -164,18 +164,26 @@ local hour = floor(os.time() / 3600)
 
 local function get_geo_city(ip_addr)
     local city = UNK_GEO
+    local city_geoname_id = nil
     local country = UNK_GEO
 
     local ok, ip = pcall(city_db.lookup, city_db, ip_addr)
-    if not ok then return city, country end
+    if not ok then return city, city_geoname_id, country end
 
     ok, city = pcall(ip.get, ip, "city", "names", "en")
     if not ok then city = UNK_GEO end
 
+    ok, city_geoname_id = pcall(ip.get, ip, "city", "geoname_id")
+    if not ok then
+        city_geoname_id = nil
+    else
+        city_geoname_id = {value = city_geoname_id, value_type = 2}
```

Comment:
@trink so like this?

Reply:
Yes, you can also save a lot of GC if you move the table outside the function and re-use it.